files/cache_utils.py

```
@@ -97,9 +99,13 @@ def get_cached_permission(cache_key: str) -> bool | None:
         result = cache.get(cache_key, version=CACHE_VERSION)
         if result is not None:
             logger.debug(f"Cache hit for key: {cache_key}")
+            record_cache_operation("permission", "get", hit=True)
+        else:
+            record_cache_operation("permission", "get", hit=False)
         return result
     except Exception as e:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)